spec/controllers/items_controller_spec.rb

```diff
@@ -4,21 +4,15 @@
   let(:wishlist) { create(:wishlist, user: controller.current_user) }
   let(:item) { create(:item, wishlist: wishlist) }
 
-  shared_examples "redirects to the 'sign_in' page" do
-    it "redirects to the 'sign_in' page" do
-      expect(response).to redirect_to(new_user_session_path)
+  shared_examples "redirects to" do |path|
+    it "redirects to the #{path}" do
+      expect(response).to redirect_to(send(path))
     end
   end
 
-  shared_examples "redirects to the wishlist page" do
-    it "redirects to the wishlist page" do
-      expect(response).to redirect_to(wishlist)
-    end
-  end
-
-  shared_examples "redirects to the wishlists page" do
-    it "redirects to the wishlists page" do
-      expect(response).to redirect_to(wishlists_url)
+  shared_examples "shows a flash message" do |flash_type|
+    it "shows a/an #{flash_type}" do
+      expect(flash[flash_type.to_s]).not_to be_nil
```

Comment:
You could make it a single `shared_examples`, pass `flash_type` as a second argument, and run its `expect` line if it's present:
```ruby
it "shows a/an #{flash_type}" do
  expect(flash[flash_type]).not_to be_nil if flash_type
end
```